lib/srv/monitor.go

```diff
@@ -233,3 +231,41 @@ func (ch trackingChannel) Write(buf []byte) (int, error) {
 	ch.t.UpdateClientActivity()
 	return n, err
 }
+
+// TrackingReadConn allows to wrap net.Conn and keeps track of the latest conn read activity.
+type TrackingReadConn struct {
+	sync.RWMutex
+	net.Conn
+	Clock      clockwork.Clock
+	lastActive time.Time
+	Ctx        context.Context
+	Cancel     context.CancelFunc
+}
```

Comment:
Since you moved this into a reusable type now, let's refactor it a bit to make a proper type like we do everywhere else i.e. have a config struct, unexport mutex, etc. Something like:

```go
type TrackingReadConnConfig struct {
  Conn    net.Conn
  Clock   clockwork.Clock
  Context context.Context
  Cancel  context.CancelFunc
}

func (c *TrackingReadConnConfig) CheckAndSetDefaults() error {
  ...
}

type TrackingReadConn struct {
  cfg        TrackingReadConnConfig
  m          sync.RWMutex
  lastActive time.Time
}

func NewTrackingReadConn(config TrackingReadConnConfig) (*TrackingReadConn, error) {
  if err := config.CheckAndSetDefaults(); err != nil {
    return nil, trace.Wrap(err)
  }
  ...
}
```